internal/resources/package/scope/scope.go

```diff
@@ -0,0 +1,79 @@
+/*
+Copyright © 2023 VMware, Inc. All Rights Reserved.
+SPDX-License-Identifier: MPL-2.0
+*/
+
+package scope
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"golang.org/x/exp/slices"
+
+	packageclustermodel "github.com/vmware/terraform-provider-tanzu-mission-control/internal/models/package/cluster"
+	commonscope "github.com/vmware/terraform-provider-tanzu-mission-control/internal/resources/common/scope"
+)
+
+// ScopedFullname is a struct for all types of source secret full names.
+type ScopedFullname struct {
+	Scope           commonscope.Scope
+	FullnameCluster *packageclustermodel.VmwareTanzuManageV1alpha1ClusterNamespaceTanzupackageMetadataPackageFullName
```

Comment:
we are just making the implementation the same for other resources, in case we need to add the new scope level then it would be easy to understand and extend, @ishangupta-ds, @ramya-bangera  do you have any thought on this